src/DeserializerFactory.php

```diff
@@ -34,9 +35,25 @@ class DeserializerFactory {
 	 */
 	private $currentFactory;
 
-	public function __construct( Deserializer $dataValueDeserializer, EntityIdParser $idParser ) {
+	/**
+	 * @var DispatchableDeserializer|null $currentEntityDeserializer
+	 */
+	private $currentEntityDeserializer;
+
+	/**
+	 * @param Deserializer $dataValueDeserializer
+	 * @param EntityIdParser $idParser
+	 * @param DispatchableDeserializer|null $currentEntityDeserializer used instead of constructing
+	 *        a new current Deserializer for entities using a current DeserializerFactory.
+	 */
+	public function __construct(
+		Deserializer $dataValueDeserializer,
+		EntityIdParser $idParser,
+		DispatchableDeserializer $currentEntityDeserializer = null
+	) {
 		$this->legacyFactory = new LegacyDeserializerFactory( $dataValueDeserializer, $idParser );
 		$this->currentFactory = new CurrentDeserializerFactory( $dataValueDeserializer, $idParser );
```

Comment:
That is not correct, since `$this->currentFactory` is also used in the `newStatementDeserializer` method below.